telegraf/1.27/entrypoint.sh

```diff
@@ -8,8 +8,6 @@ fi
 if [ $EUID -ne 0 ]; then
     exec "$@"
 else
-    # Allow telegraf to send ICMP packets and bind to privliged ports
-    setcap cap_net_raw,cap_net_bind_service+ep /usr/bin/telegraf || echo "Failed to set additional capabilities on /usr/bin/telegraf"
-
-    exec setpriv --reuid telegraf --init-groups "$@"
+    export HOME=$(getent passwd telegraf | cut -d : -f 6)
+    exec setpriv --reuid telegraf --regid telegraf --groups telegraf --bounding-set=-all,+net_raw,+net_bind_service --inh-caps=-all,+net_raw,+net_bind_service "$@"
```

Comment:
You'll still want to use the `setcap` on the binary. The `setpriv` invocation is simply setting what the invoked process is allowed to obtain (in this case, it will obtain what is in the fscaps of the telegraf binary as set by `setcap` and by `ping` in the container). `setpriv` doesn't allow setting the effective capabilities (used by the kernel in permissions checks) so we need to obtain them through some other way. I suggest:

```
if setcap cap_net_raw,cap_net_bind_service+ep /usr/bin/telegraf ; then
  export HOME=$(getent passwd telegraf | cut -d : -f 6)
  exec setpriv --reuid telegraf --regid telegraf --groups telegraf --bounding-set=-all,+net_raw,+net_bind_service --inh-caps=-all,+net_raw,+net_bind_service "$@"
else
  echo "Failed to set additional capabilities on /usr/bin/telegraf"
fi
```